sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/options/DirectoryUpdateAccessControlRecursiveOptions.java

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.datalake.options;
+
+import com.azure.core.http.rest.Response;
+import com.azure.storage.file.datalake.models.AccessControlChanges;
+import com.azure.storage.file.datalake.models.PathAccessControlEntry;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Optional parameters for Update Access Control Recursive.
+ */
+public class DirectoryUpdateAccessControlRecursiveOptions {
+    private final List<PathAccessControlEntry> accessControlList;
+    private Integer batchSize;
+    private Integer maxBatches;
+    private Consumer<Response<AccessControlChanges>> progressHandler;
+    private String continuationToken;
+    private boolean continueOnFailure;
+
+    /**
+     * Constructs a new options object.
+     * @param accessControlList The POSIX access control list for the file or directory.
+     */
+    public DirectoryUpdateAccessControlRecursiveOptions(List<PathAccessControlEntry> accessControlList) {
+        this.accessControlList = Collections.unmodifiableList(accessControlList);
```

Comment:
Should we add a null check here? or protect the call to Collections.unmodifiable list

Reply:
Yea. And I think I need to add null checks to the internal implementation method, too

Reply:
Done